### ISSUE TITLE
Allow exporter to be asynchronous via Kafka

### ISSUE THREAD

**Issue by kjozsa** [camunda/camunda#21875]
**Is your feature request related to a problem? Please describe.**
We are aiming to achieve maximum performance using Zeebe. The default exporter writes synchronously to Elasticsearch, and our team thinks we could speed this up, winning some performance gains.

**Describe the solution you'd like**
We'd like to replace the current exporter with a simple one which pushes the events to a Kafka topic. There should be another component (an "importer"?), which could read this Kafka topic, and push the messages to Elasticsearch in the exact same manner as the current exporter does. 

In other words, split the current Exporter into 2 components and add Kafka in between, to trade end-to-end latency to  performance gain.

**Comment by npepinpe** [comment]
Oh wow, long time no see :)

I think some people have played with that idea, i.e. exporting to a persisted stream, and importing there (@lenaschoenburg I think?). But this isn't a great solution for us, and instead we're moving away from a two-phase ingestion as that adds some extra delays inherently.

Is there anything preventing from writing your own exporter independently to do this? If it helps, we're focusing on improving exporting speed for Operate/Tasklist/etc. ingestion by exporting directly in the format they need, and we'll likely look at asynchronous export at that point to speed things up (though I don't think it will make a big difference, since you still don't want to overwhelm your database/secondary storage).

Of course, if you're using the ES exporter for other things, then I assume our efforts won't help directly for now.

**Comment by lenaschoenburg** [comment]
Yes, I explored a somewhat similar idea in the past but with a different motivation: reliability and isolation, not performance.

Since exporters run on isolated threads that are not shared for regular processing, I'd challenge that anything like this is actually beneficial for performance. Slow, even blocking, exporting should not slow down anything else apart from exporting itself, as long as there are enough threads available.

The only benefit I'd see is that you could remove CPU resources from Zeebe and instead assign them to this new importer that reads from Kafka and writes to Kafka. But that's just shifting things around, not improving anything.

**Comment by kjozsa** [comment]
Closed this for now based on team's comments - thanks for the feedback, we'll consider alternative approaches.
